app/models/vm_or_template.rb

```diff
@@ -472,6 +472,60 @@ def self.invoke_task_local(task, vm, options, args)
     )
   end
 
+  def self.invoke_tasks_remote_queue(options, deliver_on = nil)
+    queue_opts = {
+      :role        => "user_interface",
+      :class_name  => base_class.name,
+      :method_name => 'invoke_tasks_remote',
+      :args        => [options]
+    }
+    queue_opts[:deliver_on] = deliver_on if deliver_on
+    MiqQueue.put(queue_opts)
+  end
+
+  def self.invoke_api_task(hostname, action, ids)
+    require 'rest-client'
+    require 'json'
+
+    api_user     = Settings.webservices.remote_miq_api.user
+    api_password = MiqPassword.try_decrypt(Settings.webservices.remote_miq_api.password)
+    uri = URI::HTTPS.build(
+      :host     => hostname,
+      :userinfo => "#{api_user}:#{api_password}",
+      :path     => "/api/vms"
+    )
+
+    request = {
+      "action"    => action,
+      "resources" => ids.collect { |id| {"id" => id} }
+    }
+
+    begin
+      timeout = Settings.webservices.remote_miq_api.request_timeout
+
+      RestClient::Resource.new(
+        uri.to_s,
+        :verify_ssl   => OpenSSL::SSL::VERIFY_NONE,
+        :timeout      => timeout,
+        :open_timeout => timeout,
+        :read_timeout => timeout
+      ).post(request.to_json) do |response|
+        if [301, 302, 307].include?(response.code)
+          response.follow_redirection
+        end
+        result = JSON.parse(response)
+        if response.code >= 400
+          $log.error("Remote #{action} to #{hostname} for VMs #{ids} failed: #{result["error"]["message"]}")
+          raise
```

Comment:
@carbonin Isn't this raise going to be caught on line 523? If so doesn't err.message need to be populated.